test/helpers/setup.js

```diff
@@ -1,37 +1,32 @@
-import DestinationRequest from 'testcafe-hammerhead/lib/request-pipeline/destination-request'
 import { createTables } from '@pubsweet/db-manager'
 import config from 'config'
-import start from 'pubsweet/src/startup/start'
 import startS3Server from '@elifesciences/component-service-s3/mock'
+import db from 'pubsweet-server/src/db'
+
 // while writing tests, using the following to start the server avoids having to recompile the app
 // import {startServer: start} from 'pubsweet-server'
 
 import replaySetup from './replay-setup'
 
-let server
 let s3rver
-let serverStartAttempt = false
-
-export async function startServer() {
-  console.log('test/helpers/setup.js::startServer')
-  if (!server) {
-    if (serverStartAttempt) {
-      throw new Error(
-        "Attempting to start the server again, but a previous attempt hasn't been executed yet",
-      )
-    }
-    serverStartAttempt = true
-    // increase timeout to wait for webpack compilation
-    DestinationRequest.TIMEOUT = 180 * 1000
-    console.log('test/helpers/setup.js::startServer await start')
-    server = await start()
-    console.log('test/helpers/setup.js::startServer started')
-  }
-}
 
 export async function setup(t) {
   console.log('test/helpers/setup.js::setup')
 
+  console.log('test/helpers/setup.js::setup truncate query')
+
+  const { rows } = await db.raw(`
+    SELECT tablename
+    FROM pg_tables
+    WHERE schemaname = current_schema
+  `)
+
+  const truncateQuery = rows
+    .map(row => `TRUNCATE "${row.tablename}" CASCADE`)
```

Comment:
The `CASCADE` should take care of foreign keys dependencies, but can't say for sure without checking the schema.
Evaluate whether to wrap all the queries in a transaction? Though this should be the only change being made when `setup()` is executed.

Reply:
This is true - we should wrap this in a transaction.